miden/src/cli/run.rs

```diff
@@ -109,45 +123,99 @@ impl RunCmd {
 
 #[instrument(name = "run_program", skip_all)]
 fn run_program(params: &RunCmd) -> Result<(ExecutionTrace, [u8; 32]), Report> {
```

Comment:
It feels like this function is basically two separate functions joined with an `if` statement. I would probably separate this into something like `run_masm_program()` and `run_masp_program()`.

Then, on lines 64 - 69 above, I'd do something like:
```rust
match ext.as_str() {
    "masp" => run_masp_program(self),
    "masm" => run_masm_program(self),
    _ => return Err(Report::msg("File must have a .masm or .masp extension")),
}
```